### PR TITLE
deprecate setIdleTimeout and force it to do the same as setKeepAliveTimeSeconds

### PR DESCRIPTION
**Pull request by Sammers21**
A solution for #78

### CODE REVIEW

**Comment by Sammers21** [comment]
@ppatierno, just done.

File: src/main/java/io/vertx/mqtt/MqttClientOptions.java [hunk 1]
```diff
@@ -382,6 +382,15 @@ public MqttClientOptions setMaxMessageSize(int maxMessageSize) {
     return this;
   }
 
+  /**
+   * Do the same thing as {@link MqttClientOptions#setKeepAliveTimeSeconds(int)}. Use it instead.
+   */
+  @Deprecated
+  @Override
+  public NetClientOptions setIdleTimeout(int idleTimeout) {
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/MqttClientOptions.java]: the return type has to be `MqttClientOptions`

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java [hunk 2]
```diff
@@ -82,6 +83,7 @@
   private static final int MIN_TOPIC_LEN = 1;
   private static final String PROTOCOL_NAME = "MQTT";
   private static final int PROTOCOL_VERSION = 4;
+  private static final int DEFAULT_INTERNAL_NET_CLIENT_IDLE_TIMEOUT = 0;
```
Comment (ppatierno) [inline src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java]: I think that `DEFAULT_IDLE_TIMEOUT` is enough

File: src/test/java/io/vertx/mqtt/test/client/MqttClientConnectTest.java [hunk 3]
```diff
@@ -59,9 +59,11 @@ public void connectDisconnect(TestContext context) throws InterruptedException {
   }
 
   @Test
-  public void connectDisconnectNoOptions(TestContext context) {
+  public void connectDisconnectWithIdleOption(TestContext context) {
```
Comment (ppatierno) [inline src/test/java/io/vertx/mqtt/test/client/MqttClientConnectTest.java]: Why did you change the `connectDisconnectNoOptions` already in place ? Please create a new test for this fix.

Comment (Sammers21) [reply src/test/java/io/vertx/mqtt/test/client/MqttClientConnectTest.java]: The funny thing is that _connectDisconnectNoOptions_ is a full duplicate of _connectDisconnect_ so i have decided to modify it instead of removing as duplicate :)

Comment (ppatierno) [reply src/test/java/io/vertx/mqtt/test/client/MqttClientConnectTest.java]: ah !! You are right :-) So it's ok ... see only the other request changes I did

File: src/main/java/io/vertx/mqtt/MqttClientOptions.java (hunk 1 above)
Comment (Sammers21) [reply src/main/java/io/vertx/mqtt/MqttClientOptions.java]: Sure

File: src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java (hunk 2 above)
Comment (Sammers21) [reply src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java]: Agree

**Comment by Sammers21** [comment]
squashed
